aries_vcx_core/src/ledger/response_cacher/in_memory/mod.rs

```diff
@@ -40,7 +42,7 @@ impl ResponseCacher for InMemoryResponseCacher {
         let id = id.to_string();
         let obj = serde_json::to_string(&obj)?;
 
-        let mut cache = self.cache.lock().await;
+        let mut cache = self.cache.lock()?;
```

Comment:
There are not the same thing and, unlike other changes which really just target sequential single-threaded tests, this is production code.

The `std::sync::Mutex` is blocking and will block the entire runtime thread while waiting. The `tokio` one is designed to give back control to the runtime if the lock cannot be acquired, allowing other things to run while waiting.

Reply:
Thanks for catching this! Would been a disaster having this merged